plaidml/op/op_test.cc

```diff
@@ -32,48 +33,42 @@ Program makeProgram(const std::string& name, const std::vector<Tensor>& outputs)
   return ProgramBuilder(name, outputs).compile();
 }
 
-TEST(Op, Abs) {
-  auto I = Placeholder(DType::FLOAT32, {1, 224, 224, 3}, "I");
-  auto abs = op::abs(I);
-  auto program = makeProgram("abs", {abs});
-  IVLOG(1, program);
+TEST_F(OpTest, Abs) {
+  auto A = Placeholder(DType::FLOAT32, {10, 20});
+  auto program = makeProgram("abs", {op::abs(A)});
+  std::cout << program << std::endl;
 
-  EXPECT_THAT(program, Eq(R"#(
+  // clang-format off
+  // CHECK-LABEL: OpTest.Abs
+  // CHECK: func @abs
+  // CHECK: %[[cst:.*]] = "eltwise.sconst"() {value = 0.000000e+00 : f64} : () -> tensor<f32>
+  // CHECK: %{{.*}} = "eltwise.neg"(%{{.*}}) : (tensor<10x20xf32>) -> tensor<10x20xf32>
+  // CHECK: %{{.*}} = "eltwise.cmp_lt"(%{{.*}}, %[[cst]]) : (tensor<10x20xf32>, tensor<f32>) -> tensor<10x20xi1>
+  // CHECK: %{{.*}} = "eltwise.select"(%{{.*}}, %{{.*}}, %{{.*}}) : (tensor<10x20xi1>, tensor<10x20xf32>, tensor<10x20xf32>) -> tensor<10x20xf32>
+  // CHECK: return %{{.*}} : tensor<10x20xf32>
+  // clang-format on
 
-module {
-  func @abs(%arg0: tensor<1x224x224x3xf32> {tile.name = "I"}) -> tensor<1x224x224x3xf32> {
-    %cst = "eltwise.sconst"() {value = 0.000000e+00 : f64} : () -> f32
-    %0 = "eltwise.neg"(%arg0) : (tensor<1x224x224x3xf32>) -> tensor<1x224x224x3xf32>
-    %1 = "eltwise.cmp_lt"(%arg0, %cst) : (tensor<1x224x224x3xf32>, f32) -> tensor<1x224x224x3xi1>
-    %2 = "eltwise.select"(%1, %0, %arg0) : (tensor<1x224x224x3xi1>, tensor<1x224x224x3xf32>, tensor<1x224x224x3xf32>) -> tensor<1x224x224x3xf32>
-    return %2 : tensor<1x224x224x3xf32>
-  }
-}
-)#"));
+  runProgram(program);
 }
 
-TEST(Op, All) {
+TEST_F(OpTest, All) {
   auto I = Placeholder(DType::FLOAT32, {1, 224, 224, 3}, "I");
   auto program = makeProgram("all", {op::all(I)});
-  IVLOG(1, program);
-  EXPECT_THAT(program, Eq(R"#(
-#map0 = affine_map<() -> ()>
-#map1 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
-
+  std::cout << program << std::endl;
```

Comment:
`makeProgram` should already print this to stdout

Reply:
I assume this means I should insert the `std::cout << program << std::endl;` into `makeProgram` since it's not there yet (unlike the `makeProgram` in the edsl test)

Reply:
Oh, sorry, I didn't realize it was copied. Actually it should live in `testenv.h` and both the EDSL and Op tests should use a common function.